public/js/test/mochitest/browser_dbg_interrupts.js

```diff
@@ -0,0 +1,56 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+/**
+ * Test resuming from button and keyboard shortcuts.
+ */
+
+const TAB_URL = EXAMPLE_URL + "doc-script-switching-01.html";
+
+add_task(function*() {
+  const dbg = yield initDebugger(TAB_URL,
+    "code-script-switching-01.js",
+    "code-script-switching-02.js"
+  );
+
+  const { selectors: { getPause }, getState } = dbg;
+
+  const source1 = findSource(dbg, "code-script-switching-01.js");
+  const source2 = findSource(dbg, "code-script-switching-02.js");
+
+  yield addBreakpoint(dbg, source1.id, 5);
+  yield addBreakpoint(dbg, source2.id, 6);
+  yield addBreakpoint(dbg, source1.id, 7);
+
+  yield testBreakOnNextState(dbg);
+
+  invokeInTab("firstCall");
+  yield testKey(dbg, "resumeKey");
+  yield waitForThreadEvents(dbg, "resumed");
+  ok(!getPause(getState()), "Debugger is resumed");
+
+  yield testKey(dbg, "stepOverKey");
+  yield waitForThreadEvents(dbg, "resumed");
+  ok(!getPause(getState()), "Debugger is resumed");
+
+  yield testKey(dbg, "stepInKey");
+  yield waitForThreadEvents(dbg, "resumed");
+  ok(!getPause(getState()), "Debugger is resumed");
+
+  yield testKey(dbg, "stepOutKey");
+});
+
+const testKey = Task.async(function* (dbg, key) {
+  info("Testing Key: " + key);
+  const { selectors: { getPause }, getState } = dbg;
+  yield waitForPaused(dbg);
+  ok(getPause(getState()), "Debugger is paused");
+  pressKey(dbg, key);
+});
+
+const testBreakOnNextState = Task.async(function* (dbg) {
+  const { selectors: { getPause, getIsWaitingOnBreak }, getState } = dbg;
+  ok(!getPause(getState()), "Debugger is resumed");
+  pressKey(dbg, "pauseKey");
+  ok(getIsWaitingOnBreak(getState()), "Breaking on next execution");
+});
```

Comment:
a lot of these helpers are are there in your PR. I'll wait for that to get merged and then work on your comments. 👍